plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -2174,6 +2268,73 @@ def verify_ap_provision(self, wlc_ip_address):
 
         return provision_status, provision_details
 
+    def assign_device_to_site(self):
+        """
+        Assign the Device to the Site
+
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+
+        Returns:
+            bool: True if The device assign to the site, otherwise False.
+
+        Description:
+            Checks if the specified AP assigned to the site returns "True", otherwise logs an error
+            and returns "failed" with error details.
+        """
+        assign_network_device_to_site = {
+            'deviceIds': [self.have.get("device_id")],
+            'siteId': self.have.get("site_id"),
+        }
+        self.log("Assign device to site before update: {0} .".format(
+            assign_network_device_to_site), "INFO")
+        try:
+            response = self.dnac._exec(
+                family="site_design",
+                function='assign_network_devices_to_a_site',
+                op_modifies=True,
+                params=assign_network_device_to_site
+            )
+            self.log("Response device to site: {0} .".format(response["response"]), "INFO")
+
+            if response and isinstance(response, dict):
+                task_id = response.get("response", {}).get("taskId")
+                resync_retry_count = int(self.payload.get("dnac_api_task_timeout"))
+                resync_retry_interval = int(self.payload.get("dnac_task_poll_interval"))
+
+                while resync_retry_count:
+                    task_details_response = self.get_tasks_by_id(task_id)
+                    self.log("Status of the task: {0} .".format(self.status), "INFO")
+                    responses = {}
+                    if task_details_response.get("endTime") is not None:
+                        if task_details_response.get("status") == "SUCCESS":
+                            self.log("Task Details: {0} .".format(self.pprint(
+                                task_details_response)), "INFO")
+                            self.msg = "AP {0} Site - {1} updated Successfully".format(
+                                self.have["current_ap_config"].get("ap_name"), self.have.get("site_id"))
+                            self.log(self.msg, "INFO")
+                            return True
+                        else:
```

Comment:
else is not required, move the code out of else part.. Anyhow you break if task_details_response.get("endTime") is not None ..